engine/schema/src/com/cloud/usage/UsageVO.java

```diff
@@ -328,4 +339,48 @@ public Date getStartDate() {
     public Date getEndDate() {
         return endDate;
     }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public void setStartDate(Date startDate) {
+        this.startDate = startDate;
+    }
+
+    public void setEndDate(Date endDate) {
+        this.endDate = endDate;
```

Comment:
Since Date instances are mutable, the ``this.endDate`` should be a copy of ``endDate`` to avoid side-effects.